src/emulator/modules/SceAudioIn/SceAudioIn.cpp

```diff
@@ -29,28 +35,92 @@ EXPORT(int, sceAudioInGetMicGain) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceAudioInGetStatus) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceAudioInGetStatus, int select) {
+    if (select != SCE_AUDIO_IN_GETSTATUS_MUTE) {
+        return RET_ERROR(SCE_AUDIO_IN_ERROR_INVALID_PARAMETER);
+    }
+    return 0;
 }
 
-EXPORT(int, sceAudioInInput) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceAudioInInput, int port, void *destPtr) {
+    if (!host.audio.shared.in_port.running) {
+        return RET_ERROR(SCE_AUDIO_IN_ERROR_NOT_OPENED);
+    }
+    if (port != PORT_ID) {
+        return RET_ERROR(SCE_AUDIO_IN_ERROR_INVALID_PORT_PARAM);
+    }
+
+    while (SDL_DequeueAudio(host.audio.shared.in_port.id, destPtr, host.audio.shared.in_port.len_bytes) > 0){}
+    return 0;
 }
 
 EXPORT(int, sceAudioInInputWithInputDeviceState) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceAudioInOpenPort) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceAudioInOpenPort, SceAudioInPortType portType, int grain, int freq, SceAudioInParam param) {
+    if (host.audio.shared.in_port.running) {
+        return RET_ERROR(SCE_AUDIO_IN_ERROR_PORT_FULL);
+    }
+    if (param != SCE_AUDIO_IN_PARAM_FORMAT_S16_MONO) {
+        return RET_ERROR(SCE_AUDIO_IN_ERROR_INVALID_PORT_PARAM);
+    }
+    if (portType != SCE_AUDIO_IN_PORT_TYPE_VOICE && portType != SCE_AUDIO_IN_PORT_TYPE_RAW) {
+        return RET_ERROR(SCE_AUDIO_IN_ERROR_INVALID_PORT_TYPE);
+    }
+    if (portType == SCE_AUDIO_IN_PORT_TYPE_VOICE) {
+        if (freq != 16000) {
```

Comment:
it's more great if these constants can be constexpr to be more understandable.

Reply:
They're just samplerate value/buf size (they're not alias for something strange). How would it make it more understandable?

Reply:
DEFAULT_FREQUENCY? or something like that? Either way, It doesn't seems you want to do it, so let's just ignore

Reply:
It's not really a default frequency. You can start it with 48000 too as samplerate for a RAW port.

Reply:
I don't really understand the code, so I was just throwing an example. Naming it would make the code understandable for future contribs who want to look at it, so feel free to modify it on your will. LGTM still